asterius/src/Asterius/NFData/TH.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Asterius.NFData.TH
+  ( genNFData,
+  )
+where
+
+import Control.DeepSeq
+import Data.List (foldl1')
+import Language.Haskell.TH
+
+genNFData :: Name -> Q [Dec]
+genNFData ty = do
+  TyConI dec <- reify ty
+  case dec of
+    DataD [] ((== ty) -> True) [] Nothing cons _
+      | length cons <= 0xFF ->
```

Comment:
This is unnecessary. The reason it's there for the `Binary` TH instance generator is: we use a single byte for encoding data constructor tags, so we need to ensure the total number of datacons is no more than 256. But here we can have as many datacons as we like.